### ISSUE TITLE
Support GraphQL

### ISSUE THREAD

**Issue by wohali** [apache/couchdb#1499]
@janl:
> http://graphql.org
>
> It’s rather new, but getting good traction. This would be a nice addition to our API. Somebody might already be hacking on this ;)
>
> critique: http://kellysutton.com/2017/01/02/do-we-need-graphql.html

@davisp:
> That article also mentions JSON API that could be interesting to read up on. I skimmed the front page a bit and it seems similarly interesting to GraphQL.

@gr2m:
>I created a JSON API spec for common account tasks:
>view: http://docs.accountjsonapi.apiary.io/
>source: https://github.com/hoodiehq/account-json-api
>
>JSON API is coming from the Ember community and it has some downsides. But it’s create to just avoid all bikeshedding around REST API design and just follow a common specification. There are other similar projects. And weather it’s a first class citizen feature or some kind of plugin, I think it would be nice for CouchDB to "own" the CouchDB/JSON API story. Meaning, if someone googles "CouchDB JSON API", they should come to one of our landing pages where we describe who the two work together.
>
>The same could be for GraphQL. I think someone build a system that has a GraphQL API but also uses PouchDB somehow to make it work offline ... ? I tried to find it but no luck. ping @nolanlawson do you know of something?

@nolanlawson:
> Yes there is this: https://github.com/MikeBild/graphql-pouch

**Comment by ansarizafar** [comment]
Is there any update about Graphql support?

**Comment by wohali** [comment]
@ansarizafar nope. When there is, you'll see it linked here.

**Comment by ansarizafar** [comment]
Is there any plan to add support for graphql? Is there any Couch db tutorial for Sql developers?

**Comment by wohali** [comment]
@ansarizafar It's on the roadmap. Pull requests welcome.

We do have this in the documentation that might help you: https://docs.couchdb.org/en/stable/ddocs/views/nosql.html

**Comment by rdewolff** [comment]
It would be amazing to see GraphQL support to CouchDB!

**Comment by rdewolff** [comment]
Any update on this?

**Comment by wohali** [comment]
No. Spamming the ticket is going to get you banned from our repo.
